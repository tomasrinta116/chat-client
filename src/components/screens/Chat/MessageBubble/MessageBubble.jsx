import React from "react";
import { withStyles } from "@material-ui/core/styles";
import styles from './MessageBubble.styles';
import PropTypes from 'prop-types';

const MessageBubble = props => {
  const {
    classes: { bubbleYou, infoMessage, bubbleOther, bubbleWrapper },
    sender,
    message
  } = props;
  let bubbleClass;
  let displayMessage;
  if (sender === "You") {
    bubbleClass = bubbleYou;
    displayMessage = `${sender}: ${message}`;
  } else if (sender === "info") {
    bubbleClass = infoMessage;
    displayMessage = `${message}`;
  } else {
    bubbleClass = bubbleOther;
    displayMessage = `${sender}: ${message}`;
  }
  return (
    <div className={bubbleWrapper}>
      <div id={'displayMessage'} className={bubbleClass}>{displayMessage}</div>
    </div>
  );
};

MessageBubble.propTypes = {
  sender: PropTypes.string.isRequired,
  message: PropTypes.string.isRequired,
}

export default withStyles(styles)(MessageBubble);
