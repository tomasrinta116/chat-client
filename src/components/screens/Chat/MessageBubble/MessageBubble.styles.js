const styles = {
    bubbleWrapper: {
        width: "100%",
        display: "inline-flex", // sizes bubble to the text
        overflowWrap: "break-word",
        whiteSpace: "pre-wrap" // preserves whitespace
      },
      bubbleYou: {
        background: "#e3f2fd",
        color: "#1c54b2",
        borderRadius: 30,
        lineHeight: 1.3,
        marginBottom: "0.25rem",
        marginRight: "2rem",
        marginLeft: "auto", // pushes bubble to the right
        maxWidth: "60%",
        padding: 10
      },
      bubbleOther: {
        background: "#dbdbdb",
        color: "dark gray",
        borderRadius: 30,
        lineHeight: 1.3,
        marginBottom: "0.25rem",
        marginLeft: "2rem", // keeps bubble to the left
        maxWidth: 400,
        padding: 10
      },
      infoMessage: {
        margin: "auto",
        marginBottom: "0.5rem",
        marginTop: "0.5rem"
      },

}

export default styles; 
