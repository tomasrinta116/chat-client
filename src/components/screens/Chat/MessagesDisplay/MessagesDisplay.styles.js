const styles = {
  messageList: {
    listStyle: "none",
    padding: 0 // gets rid of the indent
  }
};

export default styles;
