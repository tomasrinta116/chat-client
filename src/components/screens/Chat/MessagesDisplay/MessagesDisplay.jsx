import React from "react";
import { withStyles } from "@material-ui/core/styles";
import MessageBubble from "../MessageBubble/MessageBubble";
import styles from './MessagesDisplay.styles';
import PropTypes from 'prop-types';

class MessagesDisplay extends React.Component {
  constructor(props) {
    super(props);
    this.newMessage = null;
    this.setNewMessageRef = element => {
      this.newMessage = element;
    };
    this.scrollToBottom = () => {
      if (this.newMessage) this.newMessage.scrollIntoView();
    };
  }

  componentDidUpdate() {
    this.scrollToBottom();
  }

  render() {
    const {
      classes: { messageList },
      messages
    } = this.props;

    return (
      <React.Fragment>
        <ul className={messageList}>
          {messages.map((messageObj, index) => {
            const { sender, message } = messageObj;
            return (
              <li key={message + index.toString()} ref={this.setNewMessageRef}>
                <MessageBubble sender={sender} message={message} />
              </li>
            );
          })}
        </ul>
      </React.Fragment>
    );
  }
}

MessagesDisplay.propTypes = {
  messages: PropTypes.arrayOf(PropTypes.object).isRequired
}

export default withStyles(styles)(MessagesDisplay);