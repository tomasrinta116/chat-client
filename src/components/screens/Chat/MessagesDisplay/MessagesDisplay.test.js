import React from "react";
import { shallow } from "enzyme";
import MessagesDisplay from "./MessagesDisplay";
import MessageBubble from "../MessageBubble/MessageBubble";

describe("<MessagesDisplay />", () => {
  let wrapper;
  const props = {
    messages: [
      { sender: "You", message: "test-you" },
      { sender: "someoneElse", message: "other-test" },
      { sender: "info", message: "someone has left" }
    ]
  };

  beforeEach(() => {
    wrapper = shallow(<MessagesDisplay {...props} />).dive();
  });

  test("does it render", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("it renders the correct number of message bubbles", () => {
    const MessageBubbles = wrapper.find(MessageBubble);
    expect(MessageBubbles).toHaveLength(3);
  });
});
