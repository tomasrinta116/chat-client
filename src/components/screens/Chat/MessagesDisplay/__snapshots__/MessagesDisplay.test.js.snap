// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessagesDisplay /> does it render 1`] = `
<Fragment>
  <ul
    className="MessagesDisplay-messageList-1"
  >
    <li
      key="test-you0"
    >
      <WithStyles(MessageBubble)
        message="test-you"
        sender="You"
      />
    </li>
    <li
      key="other-test1"
    >
      <WithStyles(MessageBubble)
        message="other-test"
        sender="someoneElse"
      />
    </li>
    <li
      key="someone has left2"
    >
      <WithStyles(MessageBubble)
        message="someone has left"
        sender="info"
      />
    </li>
  </ul>
</Fragment>
`;
